src/models/Steps.php

```diff
@@ -94,22 +98,57 @@ public function read(ContentParamsInterface $params): array
     }
 
     /**
-     * Get the current unfinished steps from experiments owned by current user
+     * Get the current unfinished steps from experiments or items owned by current user or items from team
+     *
+     * @param string $table Which table do we read from? experiments or items
+     * @param bool $team Steps from the team or only user, only for items
      */
-    public function readAll(): array
+    public function readAll(string $table, bool $team = false): array
```

Comment:
having a boolean flag in a function is a code smell and should be avoided. I realize it's easier said than done. But new code must not make the class rating go down by adding complexity:

![2021-08-12-110211_1110x486_scrot](https://user-images.githubusercontent.com/3043706/129169375-0e406638-3cd8-41f6-a553-e6385e58a19a.png)

So there should be a readAllFromTeam I think.

Reply:
Code is refactored.
That was a good exercise.
Maybe I overdid it but the addition of `getReadAllSql` and `cleanUpReadAllSQLResults` methods helps to avoid code duplication.